bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/ComfoAirCommandType.java

```diff
@@ -0,0 +1,2263 @@
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.comfoair.internal.datatypes.ComfoAirDataType;
+import org.openhab.binding.comfoair.internal.datatypes.DataTypeBoolean;
+import org.openhab.binding.comfoair.internal.datatypes.DataTypeMessage;
+import org.openhab.binding.comfoair.internal.datatypes.DataTypeNumber;
+import org.openhab.binding.comfoair.internal.datatypes.DataTypeRPM;
+import org.openhab.binding.comfoair.internal.datatypes.DataTypeTemperature;
+import org.openhab.binding.comfoair.internal.datatypes.DataTypeVolt;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Represents all valid commands which could be processed by this binding
+ *
+ * @author Holger Hees - Initial Contribution
+ * @author Hans Böhm - Refactoring
+ */
+public enum ComfoAirCommandType {
+
+    /**
+     * Below all valid commands to change or read parameters from ComfoAir
+     *
+     * @param key
+     *            command name
+     * @param data_tape
+     *            data type (can be: DataTypeBoolean.class, DataTypeMessage.class,
+     *            DataTypeNumber.class, DataTypeRPM.class,
+     *            DataTypeTemperature.class, DataTypeVolt.class)
+     * @param possible_values
+     *            possible values for write command, if it can only take certain values
+     * @param change_command
+     *            byte number for ComfoAir write command
+     * @param change_data_size
+     *            size of bytes list for ComfoAir write command
+     * @param change_data_pos
+     *            position in bytes list to change
+     * @param change_affected
+     *            list of affected commands (can be empty)
+     *            is mandatory for read-write command
+     * @param read_command
+     *            request byte number for ComfoAir read command
+     * @param read_reply_command
+     *            reply byte list size for ComfoAir read command (list of values only)
+     * @param read_reply_data_pos
+     *            list of byte positions in reply bytes list from ComfoAir
+     * @param read_reply_data_bits
+     *            byte value on read_reply_data_pos position to be considered by command (used with
+     *            DataTypeBoolean.class data_type)
+     */
+    ACTIVATE {
+        {
+            key = "bindingControl#activate";
+            data_type = DataTypeBoolean.class;
+            possible_values = new int[] { 0x03 };
+            change_command = 0x9b;
+            change_data_size = 1;
+            change_data_pos = 0;
+            change_affected = new String[] {};
+            read_command = 0x9c;
+            read_reply_command = 0x9c;
+            read_reply_data_pos = new int[] { 0 };
+            read_reply_data_bits = 0x03;
+        }
+    },
```

Comment:
I really dislike this kind of field initialization; it is easy to read but very hard to add new fields and maintain. Changing all of these to use standard constructor arguments would easily save a thousand lines of code. It is a lot of work but please do it.

Reply:
@cpmeister I'm happy to apply your suggestions to improve the code. However, I'm not quite experienced in java development, so could you please provide an example on this to ensure I get you right?